Tests/scripts/collect_tests/collect_tests.py

```diff
@@ -594,6 +595,54 @@ def _collect_pack_for_modeling_rule(
             is_nightly=is_nightly
         )
 
+    def _collect_pack_for_xsiam_component(
+        self, pack_id: str, reason_description: str, changed_file_path: Path,
+        content_item_range: Optional[VersionRange] = None, is_nightly: bool = False,
+        reason: Optional[CollectionReason] = None
+    ) -> CollectionResult:
+        """Create a CollectionResult for a pack because of an xsiam component.
+
+        Marks the pack being collected and the modeling rule that needs to be tested
+
+        Args:
+            pack_id (str): the id of the pack being collected
+            reason (Optional[CollectionReason]): the reason the pack is being collected. Defaults to None.
+            reason_description (str): the reason the pack is being collected
+            changed_file_path (Path): the path to the file that was modified
+            content_item_range (Optional[VersionRange], optional): version range. Defaults to None.
+            is_nightly (Optional[bool]): whether this is a nightly flow. Defaults to False.
+
+        Returns:
+            CollectionResult: the object detailing the pack to collect and the modeling rule that should be tested
+        """
+        if self.marketplace != MarketplaceVersions.MarketplaceV2:
+            logger.info(f'Not collecting pack {pack_id} for XSIAM component {changed_file_path} because '
+                        f'it is not a collection for an XSIAM (MarketplaceV2) marketplace - '
+                        f'marketplace is {self.marketplace}')
+            raise NothingToCollectException(changed_file_path, 'packs for XSIAM components are only collected for XSIAM')
```

Comment:
can we use an existing method to validate marketplace compatibility here?

Reply:
Unfortunately we can't since this is the problem we are fixing

In the __validate_marketplace_compatibity we have the following code:

``` 
case MarketplaceVersions.MarketplaceV2:
                # For XSIAM machines we collect tests that have not xsoar marketplace.
                # Tests for the packs that has only mpv2, or mpv2 and xpanse marketplaces,
                # will run on xsiam machines only.
                if (MarketplaceVersions.MarketplaceV2 not in content_item_marketplaces) or \
                        (MarketplaceVersions.XSOAR in content_item_marketplaces):
                    raise IncompatibleMarketplaceException(content_item_path, self.marketplace)
```

But if we have an xsiam component in the pack, we still want to install it on xsiam machines & this is what my code lets us do.

Reply:
got it, thanks. Please add a comment explaining the difference in the code, so maintainers have an easier time changing things.